src/geo/projection/globe.js

```diff
@@ -174,6 +220,116 @@ export function globeTileBounds(id: CanonicalTileID): Aabb {
     return new Aabb(bMin, bMax);
 }
 
+export function aabbForTileOnGlobe(tr: Transform, numTiles: number, z: number, x: number, y: number): Aabb {
+    const tileId = new CanonicalTileID(z, x, y);
+
+    const mx = Number.MAX_VALUE;
+    const cornerMax = [-mx, -mx, -mx];
+    const cornerMin = [mx, mx, mx];
+    const m = calculateGlobeMatrix(tr, numTiles);
+
+    if (tileId.z <= 1) {
+        // Compute minimum bounding box that fully encapsulates
+        // transformed corners of the local aabb
+        const aabb = globeTileBounds(tileId);
+        const corners = aabb.getCorners();
+
+        for (let i = 0; i < corners.length; i++) {
+            vec3.transformMat4(corners[i], corners[i], m);
+            vec3.min(cornerMin, cornerMin, corners[i]);
+            vec3.max(cornerMax, cornerMax, corners[i]);
+        }
+
+        return new Aabb(cornerMin, cornerMax);
+    }
+
+    // Find minimal aabb for a tile. Correct solution would be to compute bounding box that
+    // fully encapsulates the curved patch that represents the tile on globes surface.
+    // This can be simplified a bit as the globe transformation is constrained:
+    //  1. Camera always faces the center point on the map
+    //  2. Camera is always above (z-coordinate) all of the tiles
+    //  3. Up direction of the coordinate space (pixel space) is always +z. This means that
+    //     the "highest" point of the map is at the center.
+    //  4. z-coordinate of any point in any tile descends as a function of the distance from the center
+
+    // Simplified aabb is computed by first encapsulating 4 transformed corner points of the tile.
+    // The resulting aabb is not complete yet as curved edges of the tile might span outside of the boundaries.
+    // It is enough to extend the aabb to contain only the edge that's closest to the center point.
+    const [nw, se] = globeTileLatLngCorners(tileId);
+    const bounds = new LngLatBounds();
+    bounds.setSouthWest([nw[1], se[0]]);
+    bounds.setNorthEast([se[1], nw[0]]);
+
+    const corners = [
+        latLngToECEF(se[0], nw[1]), // south-west
+        latLngToECEF(se[0], se[1]), // south-east
+        latLngToECEF(nw[0], se[1]), // north-east
+        latLngToECEF(nw[0], nw[1])  // north-west
+    ];
+
+    // Note that here we're transforming the corners to the correct pixel space while finding the min/max values.
+    for (let i = 0; i < corners.length; i++) {
+        vec3.transformMat4(corners[i], corners[i], m);
+        vec3.min(cornerMin, cornerMin, corners[i]);
+        vec3.max(cornerMax, cornerMax, corners[i]);
+    }
+
+    if (bounds.contains(tr.center)) {
+        cornerMax[2] = 0.0;
```

Comment:
As you mentioned, for this issue of overly culling tiles:


https://user-images.githubusercontent.com/7061573/154376410-6c6d953c-7baf-4f7c-b1d7-143b9a17612b.mp4

I wonder if this should be extended to `cornerMin[2] = cornerMax[2] = 0.0`. Why are we not also collapsing the minimum to 0 on the z axis? Otherwise we may leave some negative values in here and it might be possible that this goes beyond the far plane of the frustum.

Reply:
Setting the max value to 0 alone should guarantee that the center tile will not be culled by the frustum. Actually here we preserve the same logic we had as before because it's accurate enough. I also did some debugging and have summarized it [here](https://github.com/mapbox/mapbox-gl-js/pull/11508#issuecomment-1046897429) :)